src/vm/datatypes/strings.c

```diff
@@ -530,6 +530,29 @@ static Value titleString(DictuVM *vm, int argCount, Value *args) {
     return OBJ_VAL(takeString(vm, temp, string->length));
 }
 
+static Value repeatString(DictuVM *vm, int argCount, Value *args) {
+    if (argCount != 1) {
+        runtimeError(vm, "repeat() takes one argument (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    if (!IS_NUMBER(args[1])) {
+        runtimeError(vm, "repeat() count argument must be a number");
+        return EMPTY_VAL;
+    }
+
+    ObjString *string = AS_STRING(args[0]);
+    int count = AS_NUMBER(args[1]);
+    char *temp = ALLOCATE(vm, char, (string->length * count) + 1);
+
+    strcpy(temp, string->chars);
+    while (--count > 0) {
+        strcat(temp, string->chars);
+    }
+
+    return OBJ_VAL(takeString(vm, temp, strlen(temp)));
+}
```

Comment:
Codacy is complaining about the use of `strlen` and that it could potentially cause issues if the string it's counting is not properly null terminated. In our case though, we know for definite the length of our string, so we can re-use this count in the `ALLOCATE` macro and the `takeString` function rather than re-calculating!

```suggestion
    int length = (string->length * count) + 1;
    char *temp = ALLOCATE(vm, char, length);

    strcpy(temp, string->chars);
    while (--count > 0) {
        strcat(temp, string->chars);
    }

    return OBJ_VAL(takeString(vm, temp, length - 1));
}
```